internal/rest/avi_obj_httpps.go

```diff
@@ -37,23 +39,48 @@ func (rest *RestOperations) AviHttpPSBuild(hps_meta *nodes.AviHttpPolicySetNode,
 		return nil
 	}
 	name := hps_meta.Name
-	cksum := hps_meta.CloudConfigCksum
-	cksumString := strconv.Itoa(int(cksum))
+	var httpPresentPaths []string
+	httpPresentIng := sets.NewString()
+
 	tenant := fmt.Sprintf("/api/tenant/?name=%s", hps_meta.Tenant)
 	cr := lib.AKOUser
 
 	http_req_pol := avimodels.HTTPRequestPolicy{}
 	http_sec_pol := avimodels.HttpsecurityPolicy{}
-	hps := avimodels.HTTPPolicySet{Name: &name, CloudConfigCksum: &cksumString,
+	hps := avimodels.HTTPPolicySet{Name: &name,
 		CreatedBy: &cr, TenantRef: &tenant, HTTPRequestPolicy: &http_req_pol, HTTPSecurityPolicy: &http_sec_pol}
 
+	var hppmapWithPath []nodes.AviHostPathPortPoolPG
+	var hppmapWithoutPath []nodes.AviHostPathPortPoolPG
+	var hppmapAllPaths []nodes.AviHostPathPortPoolPG
+	for _, hppmap := range hps_meta.HppMap {
+		if hppmap.Path != nil {
+			hppmapWithPath = append(hppmapWithPath, hppmap)
+			httpPresentPaths = append(httpPresentPaths, hppmap.Path[0])
+		} else {
+			hppmapWithoutPath = append(hppmapWithoutPath, hppmap)
+		}
+		httpPresentIng.Insert(hppmap.IngName)
+	}
+	sort.Slice(hppmapWithPath, func(i, j int) bool {
+		return len(hppmapWithPath[i].Path[0]) > len(hppmapWithPath[j].Path[0])
```

Comment:
Should this be a dictionary sort instead of length ? (e.g. between '/foobar' and '/index' )

Reply:
Dictionary sort will create problem for overlapping path. For paths like `/lion` and `/lionbear`, dictionary sort will sort it as `/lion, /lionbear`. Due to that, all request to `/lionbear` will be handled by `/lion` backend.